### ISSUE TITLE
I want to make a graph through this code

### ISSUE THREAD

**Issue by boonam02** [openai/gym#1531]
Hello, guys. i want to make a graph through this code. Though this code has no meaning, i need to draw the graph which can extract from this code.
So the question is how can i draw the graph??
i tried that using matplotlib.pylot but i do not know what should be put in plt.pllot( , ). 
I need all of your guys helps.
I try this
plt.xlabel('n_episodes')
        plt.ylabel('reward')
        plt.title('Learning rate')
        plt.plot('n_episodes','reward')
        plt.legend()
        plt.show()     

****
import gym
import numpy as np
import tensorflow as tf
import matplotlib.pyplot as plt

input_size = 111
n_episodes = 20
batch_size = 1
warm_up = 2
learning_rate = 0.01
momentom = 0.95
copy_steps = 50
checkpoint_path = "./save/"
loss_val = np.infty
total_max_q = 0
mean_max_q = 0.0
save_steps = 100
discount_factor = 0.98

# (state, reward, done, info)
def q_network(state ,name):
    n_neurons = (80, 56, 24)
    activation = tf.nn.elu
    initializer = tf.contrib.layers.variance_scaling_initializer
    perv_layer = state
    with tf.variable_scope(name) as scope:
        for neurons in n_neurons:
            perv_layer = tf.layers.dense(perv_layer,\
                                               neurons,\
                                               activation=activation)

    trainable_vars = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES, scope=scope.name)
    trainable_vars_by_name = {var.name[len(scope.name):]: var for var in trainable_vars}

    return perv_layer, trainable_vars_by_name

def decode_action(perv, action):
    index = action // 3
    unit = 0.3
    if (perv[index] - unit > -1): 
        if action % 3 == 0:
            perv[index] -= unit
    if (perv[index] + unit < 1):
        if action % 3 == 2:
            perv[index] += unit
    
    return perv

def e_greedy(q_values, step, n_episodes):
    e = step / (n_episodes / 2)
    if e > np.random.rand():
        return np.argmax(q_values)
    else:
        return np.random.randint(0, 24)

class ReplayMemory:
    def __init__(self, N):
        self.max = N
        self.memory = np.empty(shape=(self.max, 5), dtype=np.object)
        self.index = 0
        self.limit = 0

    def append(self, data):
        self.memory[self.index, : ] = data
        self.index = (self.index + 1) % (self.max)
        self.limit = min(self.limit + 1, self.max)

    def sample(self, batch):
        randomindex = np.random.randint(0, self.limit, size=batch)
        '''
        result = np.empty(shape=(batch, 5), dtype=np.object)
        
        for i, index in enumerate(randomindex):
            result[i, : ] = self.memory[index]
        return result
        '''
        result = self.memory[randomindex].reshape(5)

        return result[0], result[1], result[2], result[3], result[4]
    
X_state = tf.placeholder(tf.float32, shape=[None, input_size])
X_action = tf.placeholder(tf.float32, shape=[8])

online_q_values, online_vars = q_network(X_state, name="q_networks/online")
target_q_values, target_vars = q_network(X_state, name="q_networks/target")

copy_ops = [target_var.assign(online_vars[var_name]) for var_name ,target_var in target_vars.items()]
copy_online_to_target = tf.group(*copy_ops)

with tf.variable_scope("train"):
    y = tf.placeholder(tf.float32, shape=[None, 1])
    q_value = np.max(online_q_values)
    error = tf.abs(y - q_value)
    clipped_error = tf.clip_by_value(error, 0.0, 1.0)
    linear_error = 2 * (error - clipped_error)
    loss = tf.reduce_mean(tf.square(clipped_error) + linear_error)

    optimizer = tf.train.MomentumOptimizer(learning_rate, momentom, use_nesterov=True)
    training_op = optimizer.minimize(loss)

init = tf.global_variables_initializer()
saver = tf.train.Saver()

env = gym.make("Ant-v2")
replaymemory = ReplayMemory(10)

for _ in range(warm_up):
    env.reset()
    obs, reward, done, _ = env.step(np.zeros(8))
    while not done:
        perv_state = obs
        action = env.action_space.sample()
        obs, reward, done, _ = env.step(action)
        env.render()
        replaymemory.append([perv_state, action, reward, obs, 1.0 - done])
        

with tf.Session() as sess:
    init.run()
    copy_online_to_target.run()

    for step in range(n_episodes):
        env.reset()
        obs, reward, done, _ = env.step(np.zeros(8))
        action = np.array(np.zeros(8))
        while not done:
            perv_state = obs  
            perv_action = action

            q_values = online_q_values.eval(feed_dict={X_state: [obs]})
            action = e_greedy(q_values, step, n_episodes)
            print(f"ACTION : {action}")
            action = decode_action(perv_action, action)
            print(f"ACTION : {action}")
            obs, reward, done, _ = env.step(action)
            env.render()

            replaymemory.append([perv_state, action, reward, obs, 1.0 - done])

            X_state_val, X_action_val, rewards, X_next_state_val, continues = \
            (replaymemory.sample(batch_size))
            
            next_q_values = target_q_values.eval(feed_dict={X_state: [X_next_state_val]})
            max_next_q_values = np.max(next_q_values, axis=1, keepdims=True)
            y_val = rewards + continues * discount_factor * max_next_q_values

            _, loss_val = sess.run([training_op, loss], feed_dict={ \
            X_state: [X_state_val], X_action: X_action_val, y: y_val})
            

            #print(f"Step : {step:5d} | Loss : {loss_val:10f} |  ")
        if step % copy_steps == 0:
            copy_online_to_target.run()

        if step % save_steps == 0:
            saver.save(sess, checkpoint_path + f"{step}")
        
        ****

**Comment by boonam02** [comment]
the code is from ''import gym'' to ''if stepif step % save_steps == 0:
        saver.save(sess, checkpoint_path + f"{step}")"

**Comment by christopherhesse** [comment]
This doesn't really seem like a gym issue, you should try matplotlib or maybe tensorflow.
